java/src/main/java/search/BinarySearch.java

```diff
@@ -0,0 +1,30 @@
+package search;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+public class BinarySearch {
+    public <T extends Comparable> int recursiveSearch(T[] typeArray, T searchItem) throws IOException {
+        boolean present = Arrays.stream(typeArray).anyMatch(searchItem::equals);
+        if (!present) {
+            throw new IOException("Item not present in array");
+        } else {
+            int low = 0;
+            int high = typeArray.length - 1;
+            Arrays.sort(typeArray);
```

Comment:
It kind of defeats the whole purpose to sort the array in a binary search function.
If you want to find an element as quickly as possible in an array, either you assume it's sorted and you do a binary search (O(log n)), or you don't assume anything and go one by one (O(n)).
But here this function is effectively O(n log n).

Reply:
An alternative is to apply the search on a binary tree instead of an array. This way you can truly assume the data is sorted. That would require an implementation of a binary tree class though. Something to consider for future LYANL contributions.

Reply:
noted, removed in new PR